core/src/main/java/org/kiwix/kiwixmobile/core/main/ConfirmationAlertDialogFragment.java

```diff
@@ -49,14 +49,14 @@
   private SharedPreferenceUtil sharedPreferenceUtil;
   private int stringResourceId;
   private String parentDialogFragmentTAG;
-  @Inject
-  protected DialogShower alertDialogShower;
+  protected AlertDialogShower alertDialogShower;
 
   public ConfirmationAlertDialogFragment(SharedPreferenceUtil sharedPreferenceUtil,
-    String parentDialogFragmentTAG, int stringResourceId) {
+    String parentDialogFragmentTAG, int stringResourceId, Activity activity) {
     this.sharedPreferenceUtil = sharedPreferenceUtil;
     this.parentDialogFragmentTAG = parentDialogFragmentTAG;
     this.stringResourceId = stringResourceId;
+    alertDialogShower = new AlertDialogShower(activity);
```

Comment:
Nope, we use dagger to provide dependencies.

I shouldn't have allowed this constructor in the first place. Pass arguments to this fragment as a bundle and delete this constructor please.

Use dagger to inject AlertDialogShower.

Reply:
Please review

Reply:
My comment now is identical to the comment above

Reply:
@sonusourav I think i got the problem with the DI of ConfirmationAlertDialog. If you are not working on this can i send a PR.

Reply:
@sonusourav any response for @yash1ts ?

Reply:
@macgills I haven't used dagger before so exploring it took some time but got to learn something new and useful.
@yash1ts Hey it would be great if instead of opening a new PR, you can review or add your comments in this PR only. This is how we contribute and learn together in open source 😃

Reply:
Ok cool .

Reply:
I made a very long post about it in the `skillshare` channel on kiwix's slack, I don't know if it is visible to you?

The offical [docs](https://developer.android.com/training/dependency-injection) have also been updated late last year to not suck real bad

Reply:
> 
> 
> I made a very long post about it in the `skillshare` channel on kiwix's slack, I don't know if it is visible to you?
> 
> The offical [docs](https://developer.android.com/training/dependency-injection) have also been updated late last year to not suck real bad

Hey I am not there in slack. If you could add me there, if would be great. Here is my email:
`sonusouravdx001@gmail.com`

Reply:
@sonusourav https://join.slack.com/t/kiwixoffline/shared_invite/enQtOTUyMTg4NzMxMTM4LTU0MzYyZDliYjdmMDYzYWMzNDA0MDc4MWE5OGM0ODFhYjAxNWIxMjVjZTU4MTkyODJlZWFkMmQ2YTZkYTUzZDY

Reply:
> 
> 
> Nope, we use dagger to provide dependencies.
> 
> I shouldn't have allowed this constructor in the first place. Pass arguments to this fragment as a bundle and delete this constructor please.
> 
> Use dagger to inject AlertDialogShower.

Not sure if this is what you requested! Please let me know if any changes are to be made.

Reply:
This still isn't what I want.

```
  public ConfirmationAlertDialogFragment(SharedPreferenceUtil sharedPreferenceUtil,
    String parentDialogFragmentTAG, int stringResourceId) {
    this.sharedPreferenceUtil = sharedPreferenceUtil;
    this.parentDialogFragmentTAG = parentDialogFragmentTAG;
    this.stringResourceId = stringResourceId;
  }
```

This constructor should not exist, the typical looking code for passing trivial things to a Fragment is the following
```
    /**
     * Create a new instance of DetailsFragment, initialized to
     * show the text at 'index'.
     */
    public static DetailsFragment newInstance(int index) {
        DetailsFragment f = new DetailsFragment();

        // Supply index input as an argument.
        Bundle args = new Bundle();
        args.putInt("index", index);
        f.setArguments(args);

        return f;
    }
```
[source](https://developer.android.com/reference/android/app/Fragment)

Should be noted this is how you would do it when I first started writing android, the way we should share data is through a ViewModel but we don't have to worry about that here because the data is not so important.

The DialogShower is null because `.inject(this)` has not been called in this class. If you have read my rant on the skillshare channel I am sure I said something like "marking a field inject does nothing, you have to ask dagger to put an object into the fields marked inject"

You can probably delete this entire class as all it does is show a dialog, you can just use DialogShower from AddNoteDialog.